src/main/java/com/openshift/internal/restclient/DefaultClient.java

```diff
@@ -212,6 +207,18 @@ public String getResourceURI(IResource resource) {
 		}
 	}
 	
+	private RequestBody getPayload(String method, IResource payload) {
```

Comment:
@adietish this fixes my issues, but I'm wondering if it should evaluate based on the payload. I *THINK* delete may allow you to provide a payload which would allow you to define delete options (e.g. grace period), but I have not further investigated.  Do you think this is good for now? Or should we throw in cases like create (POST) where you should provide a payload not allow the library to send empty?